src/main/kotlin/platform/mixin/expression/gui/FlowStrings.kt

```diff
@@ -0,0 +1,200 @@
+import com.intellij.openapi.project.Project
+import com.llamalad7.mixinextras.expression.impl.flow.FlowValue
+import com.llamalad7.mixinextras.expression.impl.flow.expansion.InsnExpander
+import com.llamalad7.mixinextras.expression.impl.flow.postprocessing.InstantiationInfo
+import com.llamalad7.mixinextras.expression.impl.flow.postprocessing.LMFInfo
+import com.llamalad7.mixinextras.expression.impl.flow.postprocessing.MethodCallType
+import com.llamalad7.mixinextras.expression.impl.flow.postprocessing.StringConcatInfo
+import com.llamalad7.mixinextras.expression.impl.utils.ExpressionASMUtils
+import com.llamalad7.mixinextras.expression.impl.utils.FlowDecorations
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.Type
+import org.objectweb.asm.tree.ClassNode
+import org.objectweb.asm.tree.FieldInsnNode
+import org.objectweb.asm.tree.MethodInsnNode
+import org.objectweb.asm.tree.MethodNode
+import org.objectweb.asm.tree.VarInsnNode
+
+fun FlowValue.shortString(project: Project, clazz: ClassNode, method: MethodNode): String {
+    if (isComplex) {
+        return "Complex Flow (type ${type.shortName})"
+    }
+    ExpressionASMUtils.getConstant(insn)?.let { return constantString(it) }
+    ExpressionASMUtils.getCastType(insn)?.let { return castString(it) }
+    newArrayString(this)?.let { return it }
+    getDecoration<MethodCallType>(FlowDecorations.METHOD_CALL_TYPE)?.let { methodCallType ->
+        return methodString(insn as MethodInsnNode, methodCallType)
+    }
+    getDecoration<InstantiationInfo>(FlowDecorations.INSTANTIATION_INFO)?.let { info ->
+        return instantiationString(info)
+    }
+    getDecoration<StringConcatInfo>(FlowDecorations.STRING_CONCAT_INFO)?.let { _ ->
+        return "+"
+    }
+    getDecoration<LMFInfo>(FlowDecorations.LMF_INFO)?.let { info ->
+        return lmfString(info)
+    }
+    getDecoration<FlowValue>(FlowDecorations.COMPLEX_COMPARISON_JUMP)?.let { jump ->
+        return complexCmpString(insn.opcode, jump.insn.opcode)
+    }
+    return when (val insn = insn) {
+        is FieldInsnNode -> fieldString(insn)
+        is VarInsnNode -> varString(this, insn.`var`, project, clazz, method)
+        else -> opcodeString(insn.opcode) ?: longString()
+    }
+}
+
+fun FlowValue.longString(): String =
+    if (isComplex) {
+        "Complex Flow (type ${type.className})"
+    } else {
+        insn.textify()
+    }
+
+val Type.shortName get() = className.substringAfterLast('.').replace('$', '.')
+
+private fun shortOwner(owner: String) = Type.getObjectType(owner).shortName
+
+private fun shortParams(desc: String) = Type.getArgumentTypes(desc).joinToString(prefix = "(", postfix = ")") {
+    it.shortName
+}
+
+private fun constantString(cst: Any): String {
+    if (cst is Int) {
+        return when (cst) {
+            0 -> "0 or false"
+            1 -> "1 or true"
+            in Char.MIN_VALUE.code..Char.MAX_VALUE.code -> "$cst or '${cst.toChar()}'"
```

Comment:
It's probably worth escaping some chars, especially control characters and newline